packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -2331,6 +2332,23 @@ function removeEventListenerFromChild(
   return false;
 }
 // $FlowFixMe[prop-missing]
+FragmentInstance.prototype.dispatchEvent = function (
+  this: FragmentInstanceType,
+  event: Event,
+): boolean {
+  const parentHostInstance = getFragmentParentHostInstance(this._fragmentFiber);
+  if (parentHostInstance === null) {
+    if (__DEV__) {
+      console.error(
+        'You are attempting to dispatch an event on a disconnected ' +
+          'FragmentInstance. No event was dispatched.',
+      );
+    }
+    return true;
+  }
+  return parentHostInstance.dispatchEvent(event);
```

Comment:
It's interesting because we have `addEventListener` on the ref. Shouldn't those listeners be invoked too? It's kind of weird that your own listeners wouldn't fire.

We could manually invoke them but unfortunately the timing wouldn't be quite right in the capture/bubble phase.

One simple thing we can do is:

```suggestion
  if (this.__eventListeners !== null || !event.bubbles) {
      const temp = document.createTextNode('')
      // addEventListener for all the ones in __eventListeners
      parentHostInstance.appendChild(temp);
      return temp.dispatchEvent(event);
      parentHostInstance.removeChild(temp);      
  } else {
    return parentHostInstance.dispatchEvent(event);
  }
```

Most of the times the simple path is hit because you're unlike to both listen and dispatch to the same event type and even more unlikely to have non-bubbling dispatches.

Reply:
Also felt like it was a bit awkward but wasn't sure about the right semantics. This feels reasonable to me, updated to include this logic.